test/query.test.js

```diff
@@ -3945,4 +3945,30 @@ describe('Query', function() {
 
     assert.ok(users.length, 1);
   });
+
+  it('should return query helper supplied in schema options query property instead of undefined', function(done) {
+
+    const Model = db.model('Test', new Schema({
+      userName: {
+        type: String,
+        required: [true, 'userName is required']
+      }
+    }, {
+      query: {
+        byUserName(userName) {
+          return this.where({ userName });
+        }
+      }
+    }));
+
+    Model.create({ userName: 'test' }, function(error) {
+      assert.ifError(error);
```

Comment:
will this even work? We are in a callback.

```suggestion
      if (error instanceof Error) {
        return done(error);
      }
```